src/coreclr/vm/threads.cpp

```diff
@@ -2667,7 +2671,7 @@ void Thread::BaseCoUninitialize()
     STATIC_CONTRACT_GC_TRIGGERS;
     STATIC_CONTRACT_MODE_PREEMPTIVE;
 
-    _ASSERTE(GetThread() == this);
+    _ASSERTE(GetThreadNULLOk() == this);
```

Comment:
```suggestion
    _ASSERTE(GetThread() == this);
```
Assert that compares with "this" should not allow NULL

Reply:
I assumed that `this` is typically not NULL anyways so assert would be triggered. `GetThread` would be ok too.

Reply:
If both GetThread and GetThreadNULLOk work, I think we should preffer GetThread

Reply:
If the goal of the change is to speed up Checked/Debug, then `GetThreadNULLOk()` would be preferred when possible. But I think most of the gain comes from the removal of the contract stuff anyways.
On the other hand as a general recommendation as what to use, I would indeed suggest `GetThread()`, unless the code is obviously handling NULL. So from that point using `GetThread()` when possible should be preferred.

I spent some time thinking of what would be preferred in asserts, but did not feel too strongly either way and decided for `GetThreadNULLOk()`, mostly because I had to choose.
I am completely ok with ``GetThread()`